src/net.cpp

```diff
@@ -1112,6 +1095,11 @@ void CConnman::DisconnectNodes()
             }
         }
     }
+
+    for (NodeId id : disconnected_eviction_candidates) {
+        m_evictionman.RemoveCandidate(id);
+    }
```

Comment:
Is there a reason not to do this in the `Delete disconnected nodes` loop below?

Reply:
It's a different set of nodes, so different loop. Do you mean to do it in the same scope?

Reply:
I guess I don't understand why the node can't be removed from the EvictionManager either:

- in the loop above, at the point where the node is removed from `m_nodes` and added to `m_nodes_disconnected`; or
- in the loop below, at the point where the node is removed from `m_nodes_disconnected`.

Perhaps a code comment could be added to explain why.

Reply:
Cory originally suggested to it this way to untangle the removal from `m_nodes_mutex`. (See https://github.com/bitcoin/bitcoin/pull/25268#discussion_r903003875)

I think it could make sense to do it in the `Delete disconnected nodes` loop below which is also independent of `m_node_mutex`. Another option would be to do it in `PeerManagerImpl::FinalizeNode()` but that might be weird since we add candidates in `net`.

Reply:
Moved this back into the loop above.